File: words.py
import re
import time

#with open('ClarissaHarlowe.txt', encoding='ascii') as f:
with open('ClarissaHarlowe.txt', encoding='ascii') as f:
    text = f.read()
t0 = time.perf_counter()
d = {}

for word in re.split('[^a-xzA-Z]+', text):
    if len(word) > 3:
        d[word] = d.get(word, 0) + 1
max_word = max(d.items(), key=lambda x: x[1])
print(f'Most frequent: "{max_word[0]}", count: {max_word[1]}')
print(f'Elapsed: {time.perf_counter() - t0}')

File: words.cs
using System.Globalization;
using System.Text.RegularExpressions;

string text = File.ReadAllText("ClarissaHarlowe.txt");

DateTime t0 = DateTime.UtcNow;

Regex re = new Regex(@"[^a-xzA-Z]+");
Dictionary<string, int> d = new();
foreach (string word in re.Split(text))
    if (word.Length > 3)
    {
        if (d.TryGetValue(word, out int count))
            d[word] = count + 1;
        else
            d.Add(word, 1);
    }
Console.WriteLine($"Total words: {d.Count}");
var max_word = d.MaxBy(x => x.Value);
Console.WriteLine($"Most frequent: \"{max_word.Key}\", count: {max_word.Value}");
Console.WriteLine($"Elapsed: {(DateTime.UtcNow - t0).TotalSeconds.ToString(CultureInfo.InvariantCulture.NumberFormat)}");

